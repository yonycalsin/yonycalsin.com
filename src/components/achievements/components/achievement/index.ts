export { default as Achievement } from './achievement'
