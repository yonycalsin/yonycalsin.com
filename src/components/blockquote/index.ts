export { default as Blockquote } from './blockquote'
