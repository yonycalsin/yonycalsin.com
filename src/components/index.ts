export * from './achievements'
export * from './command-bar'
export * from './external-anchor'
export * from './float-socials'
export * from './illustrations'
export * from './mdx-components'
export * from './meta'
export * from './motion'
export * from './night-mode-button'
export * from './page-transition'
export * from './recommendations'
export * from './section'
export * from './work-item'
