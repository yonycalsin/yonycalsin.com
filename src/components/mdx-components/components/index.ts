export * from './anchor'
export * from './linked-heading'
export * from './table'
