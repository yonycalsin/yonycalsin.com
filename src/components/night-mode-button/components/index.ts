export * from './moon-icon'
export * from './sun-icon'
