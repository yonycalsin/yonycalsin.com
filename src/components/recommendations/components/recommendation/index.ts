export { default as Recommendation } from './recommendation'
