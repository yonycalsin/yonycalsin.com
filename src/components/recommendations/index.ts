export { default as Recommendations } from './recommendations'
