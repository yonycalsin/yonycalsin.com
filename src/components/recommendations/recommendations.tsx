import * as React from 'react'

import type { RecommendationsProps } from 'typings/components'
import { Recommendation } from './components'

function Recommendations(props: RecommendationsProps) {
  const { recommendations } = props

  return (
    <div className="grid grid-cols-1 md:grid-cols-2 gap-3 md:gap-4" role="list" aria-label="List of recommendations">
      {recommendations.map(recommendation => (
        <Recommendation key={recommendation.id} recommendation={recommendation} />
      ))}
    </div>
  )
}

export default Recommendations
