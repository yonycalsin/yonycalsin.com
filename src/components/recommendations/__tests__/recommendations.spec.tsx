import { render, screen } from '@testing-library/react'

import { featuredRecommendationsSuccess } from 'mock-server/mocks/recommendations'
import { Recommendations } from 'components'

const MOCK_RECOMMENDATIONS = featuredRecommendationsSuccess.data

const setup = () => render(<Recommendations recommendations={MOCK_RECOMMENDATIONS} />)

describe('Recommendations', () => {
  it('renders the recommendations', () => {
    const view = setup()

    const items = screen.queryAllByRole('listitem')

    expect(items).toHaveLength(MOCK_RECOMMENDATIONS.length)

    expect(view.baseElement).toMatchSnapshot()
  })
})
