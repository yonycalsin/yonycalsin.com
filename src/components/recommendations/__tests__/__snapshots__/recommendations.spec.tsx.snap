// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recommendations renders the recommendations 1`] = `
<body>
  <div>
    <div
      aria-label="List of recommendations"
      class="grid grid-cols-1 md:grid-cols-2 gap-3 md:gap-4"
      role="list"
    >
      <blockquote
        class="space-y-3 bg-white p-4 rounded-lg shadow-card border-l-4 border-primary-500"
        role="listitem"
      >
        <p>
          Yony is an amazing person at work
        </p>
        <p
          class="text-sm italic font-extrabold"
        >
          -
           
          <a
            class="font-medium text-primary-600 dark:text-primary-500 hover:underline text-gray-600"
            href="https://linkedin.com/in/mock-username"
            target="_blank"
          >
            Elon Musk
          </a>
        </p>
        <p
          class="italic text-xs text-gray-400"
        >
          CEO of Tesla, and SpaceX
        </p>
      </blockquote>
    </div>
  </div>
</body>
`;
